### DISCUSSION TITLE
Is Kong Manager (GUI) is available in the free plan?

### DISCUSSION THREAD

**Discussion by raiyansarker** [Kong/kong#8178] [Help] [answered]
Hi,
I would like to know whether Kong Manager (GUI) is available in the free plan. As per [this page](https://docs.konghq.com/gateway/), it is available in the free plan but I am not sure how to configure it. A lot of stackoverflow resources say that it is only available in the enterprise plan but those pages seems to be old.
Is it became available for free? Let me know!

**Comment by fffonion** [comment, marked as answer]
@raiyansarker You can try it for free here https://github.com/Kong/kong-manager, which includes the GUI and many more other cool features : )

**Reply by raiyansarker** [reply]
@fffonion Using this (kong/kong-gateway) image?

**Reply by raiyansarker** [reply]
@fffonion And can you elaborate a bit?

**Reply by fffonion** [reply]
> Using this (kong/kong-gateway) image?

Yes

**Reply by ocbrollingpaper** [reply]
@fffonion so OSS thingie has no manager, right? I am new to this thing, and I've never used it. Not to mention that documentation needs some rework tbh

**Reply by hbadri3** [reply]
You mean trial not free right?

**Reply by Leopoldthecoder** [reply]
I may be late but now we have Kong Manager for OSS users: https://github.com/Kong/kong-manager. It's bundled with Kong Gateway 3.4 or plus.

**Comment by ngingihy** [comment]
@fffonion there isn't any documentation on how to configure Kong manager GUI in the free plan, any insight on this?

**Reply by gAmUssA** [reply]
@ngingihy I did a stream about how to configure Kong Manager in Free plan 
https://youtu.be/lYm912ANi_E
In part two, I showed how to enable additional commercial features like vitals (not available in free version) https://youtu.be/xbFXvNUtfFQ 
Let me know how this work for you!

**Reply by akashjain971** [reply]
Kong GUI is not available in the free plan as per this snippet in docs of 2.7.x.

```
Using this guide
Throughout this guide, you will have the option to configure Kong in a few different ways. Choose your preferred method, if options are available — you don’t have to walk through all of them:
Programmatically manage Kong Gateway using its REST-based Admin API
Use the Kong Manager GUI (Not available for open-source Gateway)
Use decK for declarative configuration (YAML)
```

**Reply by gAmUssA** [reply]
Dear @akashjain971 
I believe there is confusion here - the `open-source` gateway in the text is not a __Free plan__
For example, there are two images [`kong:2.7`](https://hub.docker.com/_/kong) and [`kong/kong-gateway:2.7`](https://hub.docker.com/r/kong/kong-gateway). OSS image doesn't have Kong Manager UI. But if you use `kong-gateway` image without a license you will automatically activate the Free plan. 
I explained this in the video as well.
Hope it makes sense.

Thank you

**Reply by psyhomb** [reply]
@gAmUssA thank you for detailed explanation.

I'm using the latest `kong-gateway` image and I managed to successfully enable Kong Manager (free plan) by using [Helm chart](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml#L737-L743) (`enterprise.enabled: true` w/o `license_secret` and `manager.enabled: true`) but I don't have any control via GUI (I'm aware that dev portals, vitals are enterprise features but I also don't have access to any services, routes, consumers, plugins configuration via GUI), I guess I need to configure some permissions as well but the problem is I cannot find anything in the documentation, anything that describes how to do it.

![2022-03-01_19-03](https://user-images.githubusercontent.com/1824874/156225640-9f474340-99ba-4a8f-b8b3-ca19dbf91fb7.png)

Any help would be much appreciated.

Thank you.

**Reply by psyhomb** [reply]
Just realized I have to use some url path = kong workspace e.g. `https://<ip>/default/` in order to be able to see all the api gateway core functionalities on the left-hand side.

But still not able to use it for some reason, getting: Error: Something went wrong - also nothing useful in the logs.
It looks like Kong Manager can't access admin endpoint. 🤔 

![2022-03-01_23-27](https://user-images.githubusercontent.com/1824874/156259454-493905cd-03e9-47c2-aee7-9fb4974debb7.png)

**Reply by psyhomb** [reply]
Finally success, I also had to add following environment variables and to update `admin:` block given that we're using path based routing => `path: /admin`:
```yaml
env:
  admin_api_uri: "http://kong-proxy.mydomain.com/admin"
  admin_gui_url: "http://kong-manager.mydomain.com"

admin:
  enabled: true
  annotations:
    konghq.com/protocol: "http"

  http:
    enabled: true

  tls:
    enabled: false

  ingress:
    enabled: true
    annotations:
      kubernetes.io/ingress.class: "kong"
      konghq.com/strip-path: "true"
    path: /admin
```


![2022-03-02_14-48](https://user-images.githubusercontent.com/1824874/156374052-af613efe-1261-422f-b4d7-54b268791f03.png)

**Reply by adarsh279** [reply]
@psyhomb : I am facing the same issue with Kong Manager. Would you be able to guide me on where the configuration change need to be made? I have Kong.conf file, is that where the above mentioned changes to be done ?

**Reply by vrlorad** [reply]
Kong Manager is not available on OSS version but it is on Enterprise Free version (Enterprise without a license), so you need to use following image and of course enable enterprise features:
```
image:
  repository: kong/kong-gateway
  tag: "2.8"
```
I am able to run Kong Manager, but only default workspace is visible and there is no possibility to add new workspaces od add other user/roles as I guess RBAC is not available in Enterprise Free mode.
@gAmUssA 
Is it possible to enable Kong Manager authentication in Enterprise Free mode so it could be used only for super-admin user? I am struggling to enable authentication (Kong Manager login page) so I could use it only for super-admin user (I set admin_gui_auth to basic-auth and also set session_conf_secret according to instructions). Is Kong Manager authentication also limitation (not enabled) in Enterprise Free mode?

**Reply by vrlorad** [reply]
> Finally success, I also had to add following environment variables and to update `admin:` block given that we're using path based routing => `path: /admin`:
> 
> ```yaml
> env:
>   admin_api_uri: "http://kong-proxy.mydomain.com/admin"
>   admin_gui_url: "http://kong-manager.mydomain.com"
> 
> admin:
>   enabled: true
>   annotations:
>     konghq.com/protocol: "http"
> 
>   http:
>     enabled: true
> 
>   tls:
>     enabled: false
> 
>   ingress:
>     enabled: true
>     annotations:
>       kubernetes.io/ingress.class: "kong"
>       konghq.com/strip-path: "true"
>     path: /admin
> ```
> 
> ![2022-03-02_14-48](https://user-images.githubusercontent.com/1824874/156374052-af613efe-1261-422f-b4d7-54b268791f03.png)

@psyhomb 
Did you manage to get Kong Manager autentication enabled? Can you see login page at http://kong-manager.mydomain.com/login ?

**Reply by spirrello** [reply]
Is this possible to enable authentication in free mode?

**Reply by sumimakito** [reply]
@spirrello As per https://docs.konghq.com/gateway/latest/kong-manager/auth/basic/#set-up-basic-authentication, authentication in Kong Manger is an enterprise feature.
